### DISCUSSION TITLE
Partial results from Custom language server with vscode-languageclient

### DISCUSSION THREAD

**Discussion by SilasMarvin** [microsoft/vscode-discussions#1019] [Extension Development QnA] [answered]
I am writing a language server and a corresponding client extension for it using `vscode-languageclient`. The language server supports a number of custom commands one of which would benefit from being able to send partial results. I understand there is support in the LSP for partial results using `partial_result_token`. My main issue is a lack of anyway to receive partial results on the client. I am using the `sendRequest` method on the `LanguageClient` object to send a custom request, but this method calls my callback with only the first result from the server. Is there some way to setup a listener on the client for all incoming responses from the server? Is there a better way to listen for partial results?

Thank you for any responses and the time spent reading this!

**Comment by tamuratak** [comment, marked as answer]
VS Code doesn't support partial results. Tracked at https://github.com/microsoft/vscode/issues/105870

**Reply by SilasMarvin** [reply]
Thank you for the response! I should clarify. I know vscode doesn't support streaming, but that doesn't actually impact my use case. I'm not relying on the interaction between `vscode-languageclient` and VS Code. I want to capture the streaming results in my client extension using `vscode-languageclient`, and I'm not sure if there is a way / how to do that
